api/v1/common/sagemaker_api.go

```diff
@@ -480,3 +480,51 @@ type TransformResources struct {
 
 	VolumeKmsKeyId *string `json:"volumeKmsKeyId,omitempty"`
 }
+
+// DebugRuleConfiguration https://docs.aws.amazon.com/sagemaker/latest/dg/API_DebugRuleConfiguration.html
+type DebugRuleConfiguration struct {
+	RuleConfigurationName *string `json:"ruleConfigurationName"`
+	LocalPath             *string `json:"localPath,omitempty"`
+	// +kubebuilder:validation:Pattern=^(https|s3)://([^/]+)/?(.*)$
+	S3OutputPath       *string `json:"s3OutputPath,omitempty"`
+	RuleEvaluatorImage *string `json:"ruleEvaluatorImage"`
+	// +kubebuilder:validation:Minimum=1
+	VolumeSizeInGB *int64          `json:"volumeSizeInGB,omitempty"`
+	InstanceType   string          `json:"instanceType,omitempty"`
+	RuleParameters []*KeyValuePair `json:"ruleParameters,omitempty"`
+}
+
+// DebugHookConfig https://docs.aws.amazon.com/sagemaker/latest/dg/API_DebugHookConfig.html
+type DebugHookConfig struct {
+	LocalPath *string `json:"localPath,omitempty"`
+	// +kubebuilder:validation:Pattern=^(https|s3)://([^/]+)/?(.*)$
+	S3OutputPath             *string                    `json:"s3OutputPath"`
+	HookParameters           []*KeyValuePair            `json:"ruleParameters,omitempty"`
+	CollectionConfigurations []*CollectionConfiguration `json:"collectionConfigurations,omitempty"`
```

Comment:
Can you point me any reference ? I did not see "20" anywhere in sdk related to CollectionConfigurations

Reply:
He's referring to the [actual API](https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_DebugHookConfig.html#sagemaker-Type-DebugHookConfig-CollectionConfigurations).

That said, I am a bit hesitant to add strict SageMaker validation to our specs. I don't think that changes they make to validation like this will be communicated to us. I think we should rely on our `additional` field if the customer supplies something that doesn't fit SaegMaker validation?

Reply:
+1 thats what additional fields are for. 
Resolving this.